src/v/config/configuration.cc

```diff
@@ -355,6 +356,23 @@ configuration::configuration()
       "refreshed",
       {.visibility = visibility::tunable},
       2s)
+  , rpc_server_connection_rate_limit(
```

Comment:
I think in the final code we're only applying these to kafka (which is correct), so let's give these properties kafka-prefixed names

`kafka_connection_rate_limit` would line up nicely with the ones I'm adding elsewhere:
https://github.com/redpanda-data/redpanda/pull/3901/commits/22adbc005e6b22110bbf14fd1d5ff2f49ae0c460